generator_process/actions/prompt_to_image.py

```diff
@@ -21,12 +41,17 @@ def load_pipe(self, generator_pipeline, model, optimizations, scheduler, device)
     import torch
     import gc
 
-    use_cpu_offload = optimizations.can_use("sequential_cpu_offload", device)
-    if hasattr(self, "_cached_pipe") and self._cached_pipe[1] == model and use_cpu_offload == self._cached_pipe[2]:
-        pipe = self._cached_pipe[0]
+    invalidation_properties = (
+        action, model, device,
+        optimizations.can_use("sequential_cpu_offload", device),
+        optimizations.can_use("half_precision", device),
+    )
+    cached_pipe: CachedPipeline = self._cached_pipe if hasattr(self, "_cached_pipe") else None
+    if cached_pipe is not None and cached_pipe.is_valid(invalidation_properties):
+        pipe = cached_pipe.pipeline
     else:
         # Release the cached pipe before loading the new one.
-        if hasattr(self, "_cached_pipe"):
+        if cached_pipe is not None:
             del self._cached_pipe
```

Comment:
I believe the local reference should also be deleted.
```suggestion
            del self._cached_pipe
            del cached_pipe
```